### PR TITLE
Doc fixes

### PR DESCRIPTION
**Pull request by AlexIljin**
Some more documentation updates and fixes.

### CODE REVIEW

File: basis/threads/threads-docs.factor [hunk 1]
```diff
@@ -181,9 +181,10 @@ $nl
     "A simple thread that adds two numbers:"
     { $code "1 2 [ + . ] 2curry \"Addition thread\" spawn" }
     "A thread that counts to 10:"
+    ! Don't use $example below: it won't pass help-lint.
     { $code
       "USING: math.parser threads ;"
-      "[ 10 iota [ number>string write nl yield ] each ] \"test\" spawn"
+      "[ 10 iota [ number>string write nl yield ] each ] \"test\" spawn drop"
```
Comment (mrjbq7) [inline basis/threads/threads-docs.factor]: Why are you dropping the spawned thread?

File: basis/threads/threads-docs.factor [hunk 2]
```diff
@@ -181,9 +181,10 @@ $nl
     "A simple thread that adds two numbers:"
     { $code "1 2 [ + . ] 2curry \"Addition thread\" spawn" }
```
Comment (mrjbq7) [inline basis/threads/threads-docs.factor]: We don't drop it here...

File: basis/threads/threads-docs.factor (hunk 1 above)
Comment (AlexIljin) [inline basis/threads/threads-docs.factor]: Well, it's irrelevant to the example, and is not mentioned in the code output. I thought the missing `drop` was an oversight. Either that, or the stated output of the code.

**Comment by mrjbq7** [comment]
Merged...

**Pull request closed without merging**